llvm/lib/Target/AMDGPU/GCNSubtarget.cpp

```diff
@@ -535,6 +535,62 @@ unsigned GCNSubtarget::getMaxNumVGPRs(const MachineFunction &MF) const {
   return getBaseMaxNumVGPRs(F, MFI.getWavesPerEU());
 }
 
+bool GCNSubtarget::isRealSchedDependency(MachineInstr *DefI, int DefOpIdx,
+                                         MachineInstr *UseI,
+                                         int UseOpIdx) const {
+  if (!InstrInfo.isVOP3P(*UseI))
+    return true;
+  MachineOperand &DefOp = DefI->getOperand(DefOpIdx);
+  if (!DefOp.isReg() || !DefOp.getReg().isPhysical())
+    return true;
```

Comment:
Don't understand the isPhysical check. Why would this only apply during post ra scheduling?

Reply:
Regunits, subregs, regsOverlap, etc. are mostly defined on `MCRegister` and `Register::asMCReg()` requires a the register to be physical. That's why I thought that most of the stuff I need is only available post-ra (?)

Reply:
No, it's more important to handle virtual registers correctly which is also the easier case